llvm/lib/Analysis/BasicAliasAnalysis.cpp

```diff
@@ -333,8 +349,15 @@ struct CastedValue {
   }
 
   bool hasSameCastsAs(const CastedValue &Other) const {
-    return ZExtBits == Other.ZExtBits && SExtBits == Other.SExtBits &&
-           TruncBits == Other.TruncBits;
+    if (ZExtBits == Other.ZExtBits && SExtBits == Other.SExtBits &&
+        TruncBits == Other.TruncBits)
+      return true;
+    // If either CastedValue has a nneg zext then the sext/zext bits are
+    // interchangable for that value.
```

Comment:
But doesn't this treat them as interchangeable for *both* values? I.e. if you have `zext nneg` on one side and `zext (sext)` on the other, that's not the same right? The other side has to be just zext or just sext.